common/src/services/stateMigration.service.ts

```diff
@@ -0,0 +1,335 @@
+import { StorageService } from '../abstractions/storage.service';
+
+import { Account } from '../models/domain/account';
+import { GeneratedPasswordHistory } from '../models/domain/generatedPasswordHistory';
+import { State } from '../models/domain/state';
+import { StorageOptions } from '../models/domain/storageOptions';
+
+import { CipherData } from '../models/data/cipherData';
+import { CollectionData } from '../models/data/collectionData';
+import { EventData } from '../models/data/eventData';
+import { FolderData } from '../models/data/folderData';
+import { OrganizationData } from '../models/data/organizationData';
+import { PolicyData } from '../models/data/policyData';
+import { ProviderData } from '../models/data/providerData';
+import { SendData } from '../models/data/sendData';
+
+import { HtmlStorageLocation } from '../enums/htmlStorageLocation';
+import { KdfType } from '../enums/kdfType';
+
+// Originally (before January 2022) storage was handled as a flat key/value pair store.
+// With the move to a typed object for state storage these keys should no longer be in use anywhere outside of this migration.
```

Comment:
We should add a cut-off date where we just do a simple logout instead of maintaining this logic. I suggest adding a tech debt item and setting the expire date in half a year or something.

Reply:
https://app.asana.com/0/1201461753012676/1201507368222876